### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by kianmeng**
Found via `codespell -L crate,fromt,deques`

### CODE REVIEW

File: rayon-core/src/lib.rs [hunk 1]
```diff
@@ -560,7 +560,7 @@ impl<S> ThreadPoolBuilder<S> {
     /// to true, however, workers will prefer to execute in a
     /// *breadth-first* fashion -- that is, they will search for jobs at
     /// the *bottom* of their local deque. (At present, workers *always*
-    /// steal from the bottom of other worker's deques, regardless of
+    /// steal from the bottom of other worker's dequeues, regardless of
```
Comment (cuviper) [inline rayon-core/src/lib.rs:563]: This is not a typo -- just the plural of "deque" that you left on the line above.

File: rayon-core/src/registry.rs [hunk 2]
```diff
@@ -719,7 +719,7 @@ impl WorkerThread {
         while !latch.probe() {
             // Try to find some work to do. We give preference first
             // to things in our local deque, then in other workers
-            // deques, and finally to injected jobs from the
+            // dequeues, and finally to injected jobs from the
```
Comment (cuviper) [inline rayon-core/src/registry.rs:722]: Ditto, not a typo.

File: src/slice/mergesort.rs [hunk 3]
```diff
@@ -570,7 +570,7 @@ unsafe fn recurse<T, F>(
 
     // After recursive calls finish we'll have to merge chunks `(start, mid)` and `(mid, end)` from
     // `src` into `dest`. If the current invocation has to store the result into `buf`, we'll
-    // merge chunks from `v` into `buf`, and viceversa.
+    // merge chunks from `v` into `buf`, and vice-versa.
```
Comment (cuviper) [inline src/slice/mergesort.rs:573]: This should just be two words.

```suggestion
    // merge chunks from `v` into `buf`, and vice versa.
```

**Review by cuviper — CHANGES_REQUESTED** [review]

File: rayon-core/src/lib.rs (hunk 1 above)
Comment (cuviper) [reply rayon-core/src/lib.rs:563]: The plural possessive should be updated though...
```suggestion
    /// steal from the bottom of other workers' deques, regardless of
```

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**